proxqmox_vm_qemu read fails with 400 Bad Request when vmid already exists (provider calls /cluster/nextid during Read)

### Terraform / Provider / Environment

- **Terraform version:** `1.14.0`
- **Provider:** `Telmate/proxmox` `3.0.2-rc06`
- **OS:** Linux Ubuntu 24.04 x86_64 
- **Proxmox VE version:** 8.4.9
---

### Summary

When running `terraform plan` on an existing `proxmox_vm_qemu` resource with an explicit `vmid`, the provider fails during the **Read** phase with:

- HTTP request: `GET /api2/json/cluster/nextid?vmid=112`
- HTTP response: `400 Bad Request` with JSON body: `{"errors":{"vmid":"VM 112 already exists"}...}`

The provider appears to be calling `/cluster/nextid?vmid=<id>` while reading an already‑existing VM and treats Proxmox’s “VM X already exists” response as a hard error, causing the whole plan to fail

---

### Terraform configuration (minimal repro)

```hcl
terraform {
  required_version = ">= 1.14.0"

  backend "consul" {
    address = "my.project.com"
    scheme  = "https"
    path    = "terraform/project"
    lock    = true
    gzip    = true
  }

  required_providers {
    proxmox = {
      source  = "telmate/proxmox"
      version = "3.0.2-rc06"
    }
  }
}

provider "proxmox" {
  pm_api_url          = "https://proxmox.project.com/api2/json"
  pm_api_token_id     = "my-token-id"
  pm_api_token_secret = "my-token-secret"
  pm_tls_insecure     = false

  pm_log_enable = true
  pm_log_file   = "terraform-proxmox.log"
  pm_debug      = true

  pm_timeout = 120
}

resource "proxmox_vm_qemu" "debian13" {
  name        = "debian-terraform" # unique
  target_node = "proxmox02"
  vmid        = 112 #unique

  cpu {
    cores   = 2
    sockets = 1
    type    = "host"
  }

  memory = 2048

  disks {
    scsi {
      scsi0 {
        disk {
          storage = "local-lvm"
          size    = "32G"
        }
      }
    }

    ide {
      ide2 {
        cdrom {
          iso = "local:iso/debian-13.0.0-amd64-netinst.iso"
        }
      }
    }
  }

  network {
    id     = 0
    model  = "virtio"
    bridge = "vmbr0"
  }
}
```

State:

```bash
$ terraform state list
proxmox_vm_qemu.debian13
```

So Terraform already tracks a VM with `vmid = 112` on `target_node = proxmox02`.

---

### Steps to reproduce

1. `terraform apply` a VM with hardcoded vmid in Proxmox using the config above.

2. Ensure Terraform state contains the resource:

   ```bash
   terraform state list
   # -> proxmox_vm_qemu.debian13
   ```

3. Then run:

   ```bash
   terraform plan
   ```

---

### Observed behavior

In the provider debug log, during the **Read** of `proxmox_vm_qemu.debian13`, the provider sends:

```http
GET /api2/json/cluster/nextid?vmid=112 HTTP/1.1
Host: proxmox.project.com
...
```

Proxmox answers (VMID 112 already exists):

```http
HTTP/2.0 400 Bad Request
Content-Type: application/json; charset=UTF-8
...

{
  "errors": {
    "vmid": "VM 112 already exists"
  },
  "message": "Parameter verification failed.\n",
  "data": null
}
```

Then the provider reports an error back to Terraform:

```text
[ERROR] proxmox_vm_qemu.debian13: 400 Bad Request
[WARN] Planning encountered errors, so plan is not applyable
```

---

### Expected behavior

During a **Read** of an existing VM (`proxmox_vm_qemu`):

- The provider should **not** try to “reserve/check” the same `vmid` via `/cluster/nextid?vmid=<existing_id>`.

In other words, `terraform plan` on an already‑existing, already‑tracked VM should succeed and simply refresh the resource, not fail because the VMID is in use.

---

### Suspected cause

It looks like in the `Read` implementation for `proxmox_vm_qemu` in the 3.x provider:

- The code path that validates or obtains the VMID using `/cluster/nextid` is being executed even for **existing** resources, instead of only for `Create`.
- The provider then treats any `400` from `/cluster/nextid` as a fatal error, including the Proxmox error “VM XXX already exists”.


---

### Workarounds

For now, the only workarounds on the user side are:

- Removing the resource from state (`terraform state rm proxmox_vm_qemu.debian13`) and managing that VM outside Terraform, or
- Changing provider version / forking and patching the provider.

--- a/go.mod
+++ b/go.mod
@@ -3,7 +3,7 @@ module github.com/Telmate/terraform-provider-proxmox/v2
 go 1.24.9
 
 require (
-	github.com/Telmate/proxmox-api-go v0.0.0-20251124001151-81c14ce502d8
+	github.com/Telmate/proxmox-api-go v0.0.0-20251216222634-898857dc25c5
 	github.com/google/uuid v1.6.0
 	github.com/hashicorp/go-cty v1.5.0
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.37.0

